upickle/implicits/src-2/upickle/implicits/internal/Macros.scala

```diff
@@ -33,43 +33,13 @@ object Macros {
           s"[[${clsSymbol.name}]]. This may be due to a bug in scalac (SI-7567) " +
           "that arises when a case class within a function is upickle. As a " +
           "workaround, move the declaration to the module-level."
-        Left(msg)
-      }else{
+        fail(tpe, msg)
```

Comment:
Looking at its usage, errors are not being accumulated, and there didn’t seem to be a reason to use Either, so I modified it to throw an error immediately.